From efbc5a3414f9876dba5545147c523314499648a5 Mon Sep 17 00:00:00 2001
From: Anca Jeanina FLOAREA <anca.floarea@freescale.com>
Date: Thu, 16 May 2013 11:28:07 +0300
Subject: [PATCH 164/987] dpa_classifier: fix policer parameters for enqueue
 action

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Policer parameters overrideParams and newFqid must be
configured in order for the Policer engine to operate properly.

This patch is related to CR ENGR00262906.

Signed-off-by: Anca Jeanina FLOAREA <anca.floarea@freescale.com>
Change-Id: I4166fedb864c6ce74cd5b6051f5e9994d211918b
Reviewed-on: http://git.am.freescale.net:8181/2541
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/staging/fsl_dpa_offload/dpa_classifier.c | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier.c b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
index e337d1a..9531d33 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
@@ -2956,18 +2956,23 @@ static int action_to_next_engine_params(const struct dpa_cls_tbl_action *action,
 				next_engine_params->params.plcrParams.
 			newRelativeProfileId = (uint16_t)action->enq_params.
 				policer_params->new_rel_profile_id;
+				next_engine_params->params.plcrParams.
+					overrideParams = action->enq_params.
+					policer_params->modify_policer_params;
+				next_engine_params->params.plcrParams.
+					newFqid = action->enq_params.new_fqid;
 			} else {
 				next_engine_params->nextEngine = e_FM_PCD_DONE;
+				next_engine_params->params.enqueueParams.
+					action = e_FM_PCD_ENQ_FRAME;
+				next_engine_params->params.enqueueParams.
+					newFqid = action->enq_params.new_fqid;
+				if (action->enq_params.override_fqid)
+					next_engine_params->params.
+						enqueueParams.overrideFqid =
+						TRUE;
 			}
 
-			next_engine_params->params.enqueueParams.action =
-				e_FM_PCD_ENQ_FRAME;
-			next_engine_params->params.enqueueParams.newFqid =
-					action->enq_params.new_fqid;
-
-		if (action->enq_params.override_fqid)
-			next_engine_params->params.enqueueParams.overrideFqid =
-				TRUE;
 		if (action->enable_statistics)
 			next_engine_params->statisticsEn =
 				TRUE;
-- 
1.9.1

